debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/DumpModuleCommand.java

```diff
@@ -0,0 +1,240 @@
+		switch (args.length) {
+		case 1:
+			if (!args[0].equalsIgnoreCase("help")) {
+				filterArg = args[0];
+				status = ALL;
+			} /* else stay status.HELP */
+			break;
+		case 2:
+			filterArg = args[1];
+			switch (args[0]) {
+			case "all":
+				status = ALL;
+				break;
+			case "requires":
+				moduleFilter = DumpModuleCommand::filterModuleRequired;
+				moduleOutputter = ModularityHelper::printJ9Module;
+				status = MODULE;
+				break;
+			case "exports":
+				packageFilter = DumpModuleCommand::filterPackageModuleAndExport;
+				packageOutputter = ModularityHelper::printPackageExportVerbose;
+				status = PACKAGE;
+				break;
+			case "classes":
+				classFilter = DumpModuleCommand::filterClassByModule;
+				classOutput = ModularityHelper::printJ9Class;
+				status = CLASS;
+				break;
+			case "packages":
+				packageFilter = DumpModuleCommand::filterPackageModule;
+				packageOutputter = ModularityHelper::printJ9Package;
+				status = PACKAGE;
+				break;
+			default:
+				break; /* default stay status.HELP */
+			}
+			break;
+		default:
+			break; /* default stay status.HELP */
+		}
+		try {
+			int result;
+			J9ModulePointer modulePtr = J9ModulePointer.cast(Long.decode(filterArg));
+			J9ClassLoaderPointer classLoaderPtr = modulePtr.classLoader();;
+			switch (status) {
+			case CLASS:
+				result = ModularityHelper.iterateClassLoaderClasses(out, classFilter, classOutput, filterArg, classLoaderPtr);
+				System.out.printf("Found %d class%s\n", result, (1 == result ? "" : "es"));
+				break;
+			case MODULE:
+				result = ModularityHelper.iterateModules(out, moduleFilter, moduleOutputter, filterArg);
+				System.out.printf("Found %d module%s\n", result, (1 == result ? "" : "s"));
+				break;
+			case PACKAGE:
+				result = ModularityHelper.iterateClassLoaderPackages(out, packageFilter, packageOutputter, filterArg, classLoaderPtr);
+				System.out.printf("Found %d package%s\n", result, (1 == result ? "" : "s"));
+				break;
+			case ALL:
+				out.printf("Module:\n");
+				ModularityHelper.printJ9Module(modulePtr, out);
+				/* Run requires */
+				out.printf("Requires:\n");
+				moduleFilter = DumpModuleCommand::filterModuleRequired;
+				moduleOutputter = ModularityHelper::printJ9Module;
+				result = ModularityHelper.iterateModules(out, moduleFilter, moduleOutputter, filterArg);
+				System.out.printf("Found %d required module%s\n", result, (1 == result ? "" : "s"));
+				/* Run exports */
+				out.printf("Exports:\n");
+				packageFilter = DumpModuleCommand::filterPackageModuleAndExport;
+				packageOutputter = ModularityHelper::printPackageExportVerbose;
+				result = ModularityHelper.iteratePackages(out, packageFilter, packageOutputter, filterArg);
+				System.out.printf("Found %d exported package%s\n", result, (1 == result ? "" : "s"));
+				break;
+			default:
+				printHelp(out);
+				break;
+			}
+		} catch (CorruptDataException e) {
+			throw new DDRInteractiveCommandException(e);
+		}
+
+		return;
```

Comment:
Why the explicit return statement?

Reply:
If you're changing FindModulesCommand to adjust 'Status', then it makes sense to make this change along with it.